### PR TITLE
Import `jsonrpc` crate

### PR DESCRIPTION
**Pull request by tcharding**
Import the `rust-jsonrpc` crate from
https://github.com/apoelstra/rust-jsonrpc using current tip of master

`59646e6 Merge apoelstra/rust-jsonrpc#119: Use rust-bitcoin-maintainer-tools and re-write CI`

Full commit hash: 59646e6e6ac95f07998133b1709e4a1fa2dbc7bd

Do so using the following commands:

mkdir jsonrpc
mkdir jsonrpc/contrib
rsync -avz ../../rust-jsonrpc/master/README.md jsonrpc rsync -avz ../../rust-jsonrpc/master/src jsonrpc
rsync -avz ../../rust-jsonrpc/master/contrib/test_vars.sh jsonrpc/contrib

Then:

- Update `contrib/crates.sh` to include `jsonrpc`.
- Remove workspaces from `jsonrpc/Cargo.toml`.
- Add `jsonrpc` to repository workspace.

Finally import fuzzing, and fix up to mimic current `rust-bitcoin` setup.

### CODE REVIEW

**Comment by tcharding** [comment]
Is this an acceptable way to do this?  Still todo, bring over the fuzzing and look at the integration tests.

**Comment by tcharding** [comment]
cc @apoelstra

**Comment by apoelstra** [comment]
As a strategy for moving `jsonrpc` that works for me.

Did we agree though to move `jsonrpc`? I'd have to think a bit about whether that's what we want to do. In *theory* it's a general-purpose jsonrpc crate, though in practice we've pretty-much ignored bug reports that come from non-bitcoind users. (Although those reports are just "this doesn't support SSL" 95% of the time, and we did fix that.)

**Comment by tcharding** [comment]
> Did we agree though to move jsonrpc? 

Yeah I believe we did, remember it was the idea put together all the stuff that we feel "obligated by honour" to provide and maintain.

**Comment by apoelstra** [comment]
Ah, yeah, that makes sense. I remember that conversation but I guess it didn't occur to me that it'd mean moving `jsonrpc` into this repo.

Let's do it!

**Comment by tcharding** [comment]
We can move this repo to `rust-bitcoin` org first if you'd prefer it to not transition through my user account. That might look better, realy.

**Comment by apoelstra** [comment]
Sure -- ACked in the discussion you opened about it. But you need to do the transfer since it's coming from your org.

**Comment by tcharding** [comment]
Hey @apoelstra I'm not sure what to do about fuzzing `jsonrpc`, the fuzz tests test that rubbish read back from a socket doesn't cause us to crash, which seems like a good test but I don't think we need to run it or an hour like we do in the `rust-bitcoin` cron job. Do you have a suggestion for how we should go about running it here?

**Comment by apoelstra** [comment]
Is there any harm in running it for an hour? If we're running it on a GA runner that's probably more like 5-10 minutes on a normal computer.

But if you want to change it we could add a variable to our fuzz runner to make this configurable.

**Comment by tcharding** [comment]
Lets run it for an hour then.

File: fuzz/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# `bitcoind-json-rpc` fuzzing
+
+Currently we only fuzz `jsonrpc`, the fuzz code was imported along
+with the `jsonrpc` import from `github.com/apoelstra/rust-jsonrpc`
+commit hash: `59646e6e6ac95f07998133b1709e4a1fa2dbc7bd` which is
```
Comment (apoelstra) [inline fuzz/README.md:5]: In 53812e607438c82c6301a7ebdf4010d057cd0b02:

Trailing whitespace, and run-on sentence.

But I don't think any part of this README is necessary other than maybe the "Note to devs". People reading the code don't care what archived repo it came from, don't know who "I" is, and would probably like to know how to run the fuzzer (this information is *not* in the README as written).

Comment (tcharding) [reply fuzz/README.md:5]: Fair, will re-spin.

**Review by apoelstra — APPROVED** [review]
ACK 72388579a6e66b2d8bdc52bdc78e7c88e8f72852 successfully ran local tests

**Pull request merged**